tensilelite/Tensile/Source/lib/source/ContractionSolution.cpp

```diff
@@ -605,14 +605,19 @@ namespace Tensile
 
         for(size_t i = startStrideAB; i < a.dimensions(); i++)
         {
-            auto stride_a = problemType.sparseA ? ca.strides()[i] : a.strides()[i];
+            auto stride_a
+                = problemType.sparse && problemType.sparse != 2 ? ca.strides()[i] : a.strides()[i];
```

Comment:
condition checking should direct using "problemType.sparse == 1"
ca sould rename as compressMatrix or something else but better not "ca"